### PR TITLE
Observable comparers for SortAndBind

### PR DESCRIPTION
**Pull request by RolandPheasant**
Add  IObservable<IComparer<TObject>> option to SortAndBind to enable changing the sort order.

```cs
var comparer = SortExpressionComparer<Person>.Ascending(p => p.Name).ThenByAscending(p => p.Age);
var comparerObservable = new BehaviorSubject<IComparer<Person>>(comparer );

var sortAndBind = _cache.Connect()
            .SortAndBind(out var list, comparerObservable)
            .Subscribe()
```

This represents the completion of SortAndBind.

@dwcullop / @JakenVeina 

I know there were some previous conversations about  `SubscribeSafe`  but I skipped over that.  My question is why has the pattern  of `.SubscribeSafe(Observer.Create...` been introduced and when should it be applied?  I ask because I used `SubscribeSafe(observer)` in this PR

### CODE REVIEW

**Comment by JakenVeina** [comment]
> My question is why has the pattern of .SubscribeSafe(Observer.Create... been introduced and when should it be applied?

Bottom line is it's recommended when doing subscriptions within operators.

> Subscribes to the specified source, re-routing synchronous exceptions during invocation of the System.IObservable`1.Subscribe(System.IObserver{`0}) method to the observer's System.IObserver`1.OnError(System.Exception) channel. This method is typically used when writing query operators.

In a practical sense, I tried to write up a small test scenario that shows this in action, and after an hour, I gave up. There's a LOT of baked-in safeguards within RX that normally make `.SubscribeSafe()` redundant. I do remember this was part of the solution for #668. I also suspect that using `.SubscribeSafe()` over `.Subscribe()` has no observable effect on performance, because the meaningful difference between the two is that `.SubscribeSafe()` does 1 or 2 extra `if` checks, on top of the many that `.Subscribe()` already does. Good news is we can easily benchmark this, if we want to be sure, now that we have a few operators that use `.SubscribeSafe()` and also have benchmarks written for them.

File: src/DynamicData/Binding/SortAndBind.cs [hunk 1]
```diff
@@ -17,173 +16,233 @@ namespace DynamicData.Binding;
...
-internal sealed class SortAndBind<TObject, TKey>(
-    IObservable<IChangeSet<TObject, TKey>> source,
-    IComparer<TObject> comparer,
-    SortAndBindOptions options,
-    IList<TObject> target)
+internal sealed class SortAndBind<TObject, TKey>
     where TObject : notnull
     where TKey : notnull
 {
+    // NB: Either comparer or comparerChanged will be used, but not both.
```
Comment (JakenVeina) [inline src/DynamicData/Binding/SortAndBind.cs:23]: NB?

File: src/DynamicData/Binding/SortAndBind.cs [hunk 2]
```diff
@@ -17,173 +16,233 @@ namespace DynamicData.Binding;
...
 
-    private int GetInsertPositionBinary(TObject item)
-    {
-        var index = target.BinarySearch(item, comparer);
-        var insertIndex = ~index;
+        private int GetInsertPosition(TObject item) => options.UseBinarySearch ? GetInsertPositionBinary(item) : GetInsertPositionLinear(item);
 
-        // sort is not returning uniqueness
-        if (insertIndex < 0)
+        private int GetInsertPositionBinary(TObject item)
```
Comment (JakenVeina) [inline src/DynamicData/Binding/SortAndBind.cs:221]: Wow, this diff is obnoxiously terrible in GitHub, BTW. The syntax-aware diff in VS is so much better. Never seen a case this bad before.

**Review by JakenVeina — APPROVED** [review]

File: src/DynamicData/Binding/SortAndBind.cs (hunk 1 above)
Comment (RolandPheasant) [reply src/DynamicData/Binding/SortAndBind.cs:23]: https://www.britannica.com/dictionary/NB

Comment (JakenVeina) [reply src/DynamicData/Binding/SortAndBind.cs:23]: TIL

**Pull request merged**